syntax = "proto3";

package deeprecmaster;

message PosixFileConfig {
  repeated string input_dirs = 1;
  string output_dir = 2;
  bool auto_create_output_dir = 3;
  string input_file_meta_path = 4;
  int64 slice_size = 5;
}

message DataConfigOptions {
  int64 slice_index = 1;
  bool shuffle = 2;
  int64 shuffle_seed = 3;
  int64 num_epochs = 4;
  int64 epoch_index = 5;
}

message DataConfig {
  enum DataSourceType {
    POSIX_FILE = 0;
  }
  enum DataConsumerType {
    TRAINING = 0;
  }
  string config_name = 1;
  DataConsumerType consumer = 2;
  DataSourceType dstype = 3;
  PosixFileConfig file_config = 4;
  DataConfigOptions options = 5;
}
